pkg/manifests/manifests.go

```diff
@@ -1424,6 +1425,29 @@ func (f *Factory) PrometheusK8s(grpcTLS *v1.Secret, telemetrySecret *v1.Secret)
 	for k := range p.Spec.RemoteWrite {
 		rw := &p.Spec.RemoteWrite[k]
 
+		// Fixes: https://issues.redhat.com/browse/OCPBUGS-38289
+		// The code is condensed here to simplify backporting.
+		// This is a temporary workaround because the NoProxy field isn't exported on the Prometheus CRD in older versions.
+		// If the cluster proxy or remoteWrite config change, this code will re-run, ensuring no stale state.
+		// The httpproxy package used here is the same as used by Prometheus, which helps with consistency.
+		if f.proxy.HTTPSProxy() == "" && f.proxy.HTTPProxy() == "" {
```

Comment:
(nit) we could as well avoid the early continue and always call ProxyFunc()?

Reply:
I wanted the added logic to be the least intrusive possible.
I adjusted it